engine/src/main/java/com/datatorrent/stram/engine/StreamingContainer.java

```diff
@@ -776,6 +776,26 @@ public void processHeartbeatResponse(ContainerHeartbeatResponse rsp)
         bufferServer.purge(lastCommittedWindowId - 1);
       }
 
+      for (Node<?> node : nodes.values()) {
+
+        if (!node.context.stateless) {
+          StorageAgent storageAgent = node.context.getValue(OperatorContext.STORAGE_AGENT);
+          try {
+            for (long windowId : storageAgent.getWindowIds(node.getId())) {
```

Comment:
sort by windowId and break out of the loop once windowId >= lastCommittedWindowId